spec/features/courses/results_spec.rb

```diff
@@ -0,0 +1,144 @@
+          provider_name: "ACME SCITT A0",
+          provider_code: "T92",
+          website: "https://scitt.org",
+          address1: "Building 64",
+          address2: "32 Copton Lane",
+          address3: "Bracknel",
+          address4: "Berkshire",
+          postcode: "NXT STP")
+  end
+
+  let(:course) do
+    build(:course,
+          name: "Primary",
+          course_code: "X130",
+          provider: provider,
+          provider_code: provider.provider_code,
+          recruitment_cycle: current_recruitment_cycle)
+  end
+
+  let(:current_recruitment_cycle) { build :recruitment_cycle }
+
+  let(:results_page) { PageObjects::Page::Results.new }
+
+  let(:subject) do
+    build(:subject,
+          :english,
+          scholarship: "2000",
+          bursary_amount: "4000",
+          early_career_payments: "1000")
+  end
+
+  let(:accrediting_provider) { build(:provider) }
+  let(:decorated_course) { course.decorate }
+  let(:courses) { [course] }
+  let(:courses_request) do
+    stub_api_v3_request(
+      "/recruitment_cycles/#{Settings.current_cycle}/providers/courses?fields%5Bcourses%5D=provider_code,course_code,name,description,funding_type,provider,accrediting_provider&fields%5Bproviders%5D=provider_name,address1,address2,address3,address4,postcode&include=provider,accrediting_provider&page%5Bpage%5D=1&page%5Bper_page%5D=10",
+      resource_list_to_jsonapi(courses, include: %i[provider accrediting_provider], links: {
+        last: "/recruitment_cycles/#{Settings.current_cycle}/providers/courses?fields%5Bcourses%5D=provider_code,course_code,name,description,funding_type,provider,accrediting_provider&fields%5Bproviders%5D=provider_name,address1,address2,address3,address4,postcode&include=provider,accrediting_provider&page%5Bpage%5D=1&page%5Bper_page%5D=10",
+      }),
+    )
+  end
+  let(:page_index) { nil }
+
+  before do
+    courses_request
+
+    visit results_path(page: page_index)
+  end
+
+  it "Requests the courses" do
+    expect(courses_request).to have_been_requested
+  end
+
+  context "multiple courses" do
+    let(:course1) do
+      build(:course,
+            name: "First Course",
+            course_code: "X130",
+            provider: provider,
+            provider_code: provider.provider_code,
+            recruitment_cycle: current_recruitment_cycle,
+            accrediting_provider: provider,
+            description: "PGCE with QTS",
+            funding_type: "salary")
+    end
+
+    let(:course2) do
+      build(:course,
+            name: "Second Course",
+            course_code: "X255",
+            provider: provider2,
+            provider_code: provider.provider_code,
+            recruitment_cycle: current_recruitment_cycle,
+            description: "QTS",
+            funding_type: "salary")
+    end
+    let(:courses) { [course1, course2] }
+
+    it "Lists all courses" do
+      expect(results_page.courses.first.name.text).to eq("#{courses.first.name} (#{courses.first.course_code})")
```

Comment:
Feel like it's worth moving some of this out to a method/some out to variables - hard to see what this is doing assertions against.

Varaibles could be `first_course`/`second_course`
Method could be `expect_course_to_appear_on_results_page(first_course`)